envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/config/LocalReplyConfigFactory.kt

```diff
@@ -184,25 +190,20 @@ class LocalReplyConfigFactory(
             validateHeaderMatcher(matcherAndMapper.headerMatcher)
         }
 
-        if (definitions != 1) {
-            throw IllegalArgumentException(
-                "One and only one of: headerMatcher, responseFlagMatcher, statusCodeMatcher has to be defined.")
+        require(definitions == 1) {
+            "One and only one of: headerMatcher, responseFlagMatcher, statusCodeMatcher has to be defined."
         }
     }
 
     private fun validateHeaderMatcher(headerMatcher: HeaderMatcherProperties) {
-        if (headerMatcher.exactMatch.isNotEmpty() && headerMatcher.regexMatch.isNotEmpty()) {
-            throw IllegalArgumentException(
-                "Only one of: exactMatch, regexMatch can be defined."
-            )
+        require(headerMatcher.exactMatch.isEmpty() || headerMatcher.regexMatch.isEmpty()) {
```

Comment:
why you changed this checks to require?

Reply:
detekt failed with information that using require is more idiomatic to kotlin